### PR TITLE
Better explain role of existing requirements.txt

### PR DESCRIPTION
**Pull request by mikepqr**
This commit adds an explanation of the role of existing requirements.txt
files to the README.

I found this very unclear and opened #1358 because of my confusion. The
existing short note that requirements.txt "might interfere" didn't
really help me. I'm not sure how common my confusion is, but I hope my
changes makes things clearer to new users (assuming what I've written is
in fact correct!)

(In addition to my additions, I moved the "Updating requirements"
section up in the doc, since it seems like a core workflow.)

**Changelog-friendly one-liner**: 

##### Contributor checklist

- [ ] Provided the tests for the changes.
- [X] Gave a clear one-line description in the PR (that the maintainers can add to CHANGELOG.md on release).
- [x] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -62,10 +61,6 @@ project so conditional dependencies that require a specific Python version,
 or other environment markers, resolve relative to your project's
 environment.
 
-**Note**: ensure you don't have ``requirements.txt`` if you compile
-``setup.py`` or ``requirements.in`` from scratch, otherwise, it might
-interfere.
-
```
Comment (atugushev) [inline README.rst:68]: Could you elaborate on these deletions above? There is a reason for this note, see https://github.com/jazzband/pip-tools/issues/744.

Comment (mikepqr) [reply README.rst:68]: I found "from scratch" and "might interfere" very vague ("from scratch" in what sense? interfere how? why only "might"?).

I think the new text covers how an existing requirements.txt affects pip-compile better in the sense that explains what will happen more precisely and completely. But I'm happy to restore it if you prefer?

Comment (atugushev) [reply README.rst:68]: > I found "from scratch" and "might interfere" very vague ("from scratch" in what sense? interfere how? why only "might"?).

I'd prefer to keep the note. I like the explanation in the "Updating requirements" section.  Would you like to improve the note somehow?

Comment (mikepqr) [reply README.rst:68]: How about "Note: To compile setup.py or requirements.in from scratch, you must first delete any existing requirements.txt file"?

Comment (atugushev) [reply README.rst:68]: That's a good direction. Perhaps we should clarify shortly why `requirements.txt` must be deleted.

Comment (mikepqr) [reply README.rst:68]: Something like this (basically copy/pasted from the new text below)?

"Note: If ``pip-compile`` finds an existing ``requirements.txt`` file, and that file fulfils the dependencies, then no changes will be made, even if updates are available. To compile setup.py or requirements.in from scratch, first delete any existing requirements.txt file, or see "Updating requirements" for alternative approaches."

Comment (mikepqr) [reply README.rst]: Pushed another revision. Please let me know if this is more along the right lines.

Comment (atugushev) [reply README.rst]: Thanks! Looks good 👍🏻

**Review by atugushev — APPROVED** [review]
LGTM 👍

**Comment by atugushev** [comment]
Thanks @mikepqr 🙏🏻

**Comment by mikepqr** [comment]
Thank you for your help!

**Pull request merged**